### PR TITLE
feat: add param for gas per message byte

### PR DESCRIPTION
**Pull request by rahulghangas**
Adds `gasPerMsgByte` as params to the payment module. 
Defines relevant stores and queries

Relevant changes in 
`proto/payment/*`
`x/payment/keeper*`
`x/payment/types/*`

- [x] finished 1st part of #949

Note: To be merged after #893 and constants for default values can be defined in a cumulative PR

### CODE REVIEW

**Comment by SweeXordious** [comment]
I guess this PR depends on this one https://github.com/celestiaorg/celestia-app/pull/893, right?

**Comment by rahulghangas** [comment]
> I guess this PR depends on this one https://github.com/celestiaorg/celestia-app/pull/893, right?

Yes, a review makes sense after #893 is merged

**Comment by evan-forbes** [comment]
I know we'll merge #893 soon, but converting to a draft until then as this PR has many duplicate changes and shouldn't be reviewed or merged until its predecessor is

File: x/blob/types/params.go [hunk 1]
```diff
@@ -108,3 +117,17 @@ func validateMinMaxSquareSizeOrder(minSquareSize, maxSquareSize uint32) error {
...
+
+// validateGasPerMsgByte validates the GasPerMsgByte param
+func validateGasPerMsgByte(v interface{}) error {
+	gasPerMsgByte, ok := v.(uint32)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", v)
+	}
+
+	if gasPerMsgByte == 0 {
+		return fmt.Errorf("max square size cannot be 0")
```
Comment (SweeXordious) [inline x/blob/types/params.go:129]:
```suggestion
		return fmt.Errorf("gas per message byte cannot be 0")
```

**Review by SweeXordious — APPROVED** [review]
utAck. 
Left an error message fix.

File: proto/blob/params.proto [hunk 2]
```diff
@@ -11,4 +11,5 @@ message Params {
 
   uint32 min_square_size = 1 [(gogoproto.moretags) = "yaml:\"min_square_size\""];
   uint32 max_square_size = 2 [(gogoproto.moretags) = "yaml:\"max_square_size\""];
+  uint32 gas_per_msg_byte = 3 [(gogoproto.moretags) = "yaml:\"gas_per_msg_byte\""];
```
Comment (evan-forbes) [inline proto/blob/params.proto:14]: [can be done in a followup]
we should probably call this blob instead of msg

**Review by evan-forbes — COMMENTED** [review]
left one comment that could be addressed later, otherwise LGTM

**Comment by rahulghangas** [comment]
Followup changes in PR that addresses #1001

**Review by evan-forbes — APPROVED** [review]
utACK LGTM

**Pull request merged**